website_sale_charge_payment_fee/controllers/website_sale.py

```diff
@@ -25,7 +25,9 @@ def payment(self, **post):
                     'payment.acquirer'].browse(int(payment_fee_id))
                 values['selected_acquirer'] = selected_acquirer
             else:
-                selected_acquirer = values['acquirers'][0]
+                selected_acquirer = request.env['payment.acquirer'].browse()
+                if values['acquirers']:
+                    selected_acquirer = values['acquirers'][0]
```

Comment:
I don't know this module but I'm a bit sceptic about this change. The original module provides a list of rendered HTML, hence what you would have here is not an ID or a recordset and the template does seem to take this into account.
Am I wrong?

Reply:
Hi, thanks for checking!
Actually the original module provides a list of records found with `search` https://github.com/odoo/odoo/blob/10.0/addons/website_sale/controllers/main.py#L744-L746
and then appended one by one to `value['acquirers]` in https://github.com/odoo/odoo/blob/10.0/addons/website_sale/controllers/main.py#L760; the rendered HTML is just assigned to the `button` field of every acquirer in the list.

Reply:
hmm seems you are right. Sorry, I probably checked too quickly :sweat: . Then I don't see any blockers...